client/src/components/PortfolioGraph.tsx

```diff
@@ -3,55 +3,163 @@ import Cytoscape from "cytoscape";
 import CytoscapeComponent from "react-cytoscapejs";
 // @ts-ignore
 import fcose from "cytoscape-fcose";
-import { RawPortfolioGraphJson } from "./APIDataTypes";
+import { AddressRecord, PortfolioGraphNode, RawPortfolioGraphJson } from "./APIDataTypes";
+import { withMachineInStateProps } from "state-machine";
+import helpers from "util/helpers";
 
 Cytoscape.use(fcose);
 
-const formatGraphJSON = (rawJSON: RawPortfolioGraphJson): cytoscape.ElementDefinition[] => {
-  const nodes: cytoscape.ElementDefinition[] = rawJSON.nodes.map((node) => ({
+const layout = {
+  name: "fcose",
+  nodeDimensionsIncludeLabels: true,
+  animate: false,
+  quality: "proof",
+  idealEdgeLength: 200,
+  nodeSeparation: 300,
+};
+
+function createNode(id: string, type: string, value: string): cytoscape.NodeDefinition {
+  return {
     group: "nodes",
     data: {
-      id: node.id.toString(),
-      type: node.value.kind,
-      value: node.value.value,
+      id,
+      type,
+      value,
     },
-  }));
+  };
+}
 
-  const edges: cytoscape.ElementDefinition[] = rawJSON.edges.map((edge) => ({
+function createEdge(
+  source: string,
+  target: string,
+  count_hpd_registrations: number = 1
+): cytoscape.EdgeDefinition {
+  return {
     group: "edges",
     data: {
-      source: edge.from.toString(),
-      target: edge.to.toString(),
-      count_hpd_registrations: edge.reg_contacts,
+      source,
+      target,
+      count_hpd_registrations,
     },
-  }));
+  };
+}
+
+/**
+ * Generates one node for the search BBL and one for the detail BBL to show on the portfolio graph.
+ */
+function generateAdditionalNodes(searchAddr: AddressRecord, detailAddr?: AddressRecord) {
+  var additionalNodes = [];
+  const searchBBLNode = createNode(
+    "searchaddr",
+    "searchaddr",
+    `${searchAddr.housenumber} ${searchAddr.streetname}, ${searchAddr.boro}`
+  );
+  additionalNodes.push(searchBBLNode);
+
+  if (detailAddr) {
+    const detailBBLNode = createNode(
+      "detailaddr",
+      "detailaddr",
+      `${detailAddr.housenumber} ${detailAddr.streetname}, ${detailAddr.boro}`
+    );
+    additionalNodes.push(detailBBLNode);
+  }
+  return additionalNodes;
+}
+
+/**
+ * Given an existing array of graph nodes and an Address Record, this function
+ * finds the `id` property of a node of type `name` that corresponds to the owner
+ * (or owners) listed with the address.
+ */
+export function getLandlordNodeIDFromAddress(
```

Comment:
any chance you got to do some profiling of this on larger sets? how does the performance seem?

Reply:
lemme check!

Reply:
Ok did a quick profile and it seems like the computation is pretty instant even for our largest portfolio! Check out this cute pic too and see if u can spot our new bbls: 
![image](https://user-images.githubusercontent.com/12834575/135516654-799ee235-277b-44c2-9dc8-894aefa53ae1.png)

Reply:
🤩 bb's first bbls 🤩